borg/helpers.py

```diff
@@ -146,8 +159,18 @@ def load(cls, repository, key=None, force_tam_not_required=False):
             if not manifest_required and security_required:
                 logger.debug('Manifest is TAM verified and says TAM is *not* required, updating security database...')
                 os.unlink(tam_required_file(repository))
+        manifest.check_repository_compatibility(operation)
         return manifest, key
 
+    def check_repository_compatibility(self, operation):
+        if operation == REPO_OPERATION_NO_CHECK:
+            return
+        if operation not in self.config:
+            return
+        requirements = self.config[operation]
```

Comment:
~ maybe use a separate dict in `config` for this, like `feature_flags` or so, to keep the `config` namespace clean.